File: September 2024/September 4/taumBday.js
function taumBday(b, w, bc, wc, z) {
    let BIb = BigInt(b)
    let BIw = BigInt(w)
    let BIbc = BigInt(bc)
    let BIwc = BigInt(wc)
    let BIz = BigInt(z)
    let blackCost = 0
    let whiteCost = 0
  
    if((wc+z)<bc){
      blackCost = BigInt((BIwc+BIz) * BIb)
    }else{
      blackCost = BigInt(BIb*BIbc)
    }
    
    if ((bc+z)<wc){
      whiteCost = BigInt((BIbc+BIz) * BIw)
    } else{
      whiteCost = BigInt(BIw*BIwc)
    }
    
    return BigInt(blackCost + whiteCost)
  
  }
  

File: September 2024/September 4/taum_bday.py
def taum_bday(b, w, bc, wc, z):
    black_cost = 0
    white_cost = 0

    if wc+z < bc:
        black_cost = (wc+z) * b
    else:
        black_cost = bc * b

    if bc+z < wc:
        white_cost = (bc+z) * w
    else:
        white_cost = wc * w

    return black_cost + white_cost